public/src/components/epicTests/epicTestsForm.tsx

```diff
@@ -138,26 +165,84 @@ class EpicTestsForm extends React.Component<EpicTestFormProps, EpicTestsFormStat
       })
   }
 
+  requestEpicTestsLock = () => {
+    requestLock(FrontendSettingsType.epicTests).then(response =>
+      response.ok ? this.fetchStateFromServer() : alert("Error - can't request lock!")
+    );
+  }
+
+  requestTakeControl = () => {
+    requestTakeControl(FrontendSettingsType.epicTests).then(response =>
+      response.ok ? this.fetchStateFromServer() : alert("Error - can't take back control!")
+    );
+  }
+
+  renderButtonsBar = () => {
+    const {classes} = this.props;
+    if (!this.state.editMode) {
+      if (this.state.lockStatus.locked) {
+        const friendlyName: string | undefined = this.state.lockStatus.email && this.makeFriendlyName(this.state.lockStatus.email);
+        const friendlyTimestamp: string | undefined = this.state.lockStatus.timestamp && this.makeFriendlyDate(this.state.lockStatus.timestamp);
+        return (
+          <>
+            <Typography className={classes.warning}>File locked for editing by {friendlyName} (<a href={"mailto:" + this.state.lockStatus.email}>{this.state.lockStatus.email}</a>) at {friendlyTimestamp}</Typography>
+            <div className={classes.buttons}>
+              <ButtonWithConfirmationPopup
+                buttonText="Take control"
+                confirmationText={`Are you sure? Please tell ${friendlyName} that their unpublished changes will be lost.`}
+                onConfirm={this.requestTakeControl}
+                icon={<LockOpenIcon />}
+              />
+            </div>
+          </>
+        );
+      } else {
+        return (
+          <Button
+            onClick={this.requestEpicTestsLock}
+            variant="contained" color="primary"
+          >Edit</Button>
+        )
+      }
+    } else {
+      return (
+        <div>
+          <ButtonWithConfirmationPopup
+          buttonText="Publish"
+          confirmationText="Are you sure? This will replace all live tests!"
+          onConfirm={this.save}
+          icon={<CloudUploadIcon />}
+        />
+        <ButtonWithConfirmationPopup
+        buttonText="Cancel"
+        confirmationText="Are you sure? All unpublished data will be lost!"
+        onConfirm={() => this.cancel()}
+        icon={<RefreshIcon />}
+        />
+        </div>
+        )
+    }
+  }
+
+  makeFriendlyDate = (timestamp: string): string => {
+    const datetime = new Date(timestamp);
+    return `${datetime.getHours()}:${datetime.getMinutes()} on ${datetime.getDate()}/${datetime.getMonth() + 1}/${datetime.getFullYear()}`;
+  }
+
+  makeFriendlyName = (email: string): string => {
+    const firstName = email.split('.')[0];
+    const lastName = email.split('.')[1].split('@')[0];
```

Comment:
an alternative to this is to use `match`, e.g.
`'first.last@example.com'.match(/^([a-z]*)\.([a-z]*)@.*/)`
Which returns `[ "first.last@example.com", "first", "last" ]`

Although regex can be quite unreadable! I don't have a strong preference, what do you think?

Reply:
Don't forget we have x.y.freelancer and x.y.contractor too!

Reply:
Oh yes!

Reply:
good point! Perhaps it's more useful just to display the email address then?

Reply:
otherwise this should work:
`/^([a-z]*)\.([a-z]*)(\.[a-z]*)?@.*/`

Reply:
Ah yes - was just testing this which also seems to work /^([a-z]*)\.([a-z]*).*@.*/

Reply:
Have updated with regex in latest commit.